x-pack/plugins/maps/public/classes/styles/vector/components/symbol/icon_preview.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+  CUSTOM_ICON_PIXEL_RATIO,
+  createSdfIcon,
+  // @ts-expect-error
+} from '../../symbol_utils';
+
+export interface Props {
+  svg?: string;
+  cutoff?: number;
+  radius?: number;
+  isSvgInvalid?: boolean;
+}
+
+interface State {
+  map: MapboxMap | null;
+  iconColor: string | null;
+}
+
+export class IconPreview extends Component<Props, State> {
+  static iconId = `iconPreview`;
+  private _checker?: ResizeChecker;
+  private _isMounted = false;
+  private _containerRef: HTMLDivElement | null = null;
+
+  state: State = {
+    map: null,
+    iconColor: '#E7664C',
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._initializeMap();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.svg !== prevProps.svg ||
+      this.props.cutoff !== prevProps.cutoff ||
+      this.props.radius !== prevProps.radius
+    ) {
+      this._syncImageToMap();
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    if (this._checker) {
+      this._checker.destroy();
+    }
+    if (this.state.map) {
+      this.state.map.remove();
+      this.state.map = null;
+    }
+  }
+
+  _setIconColor = (iconColor: string) => {
+    this.setState({ iconColor }, () => {
+      this._syncPaintPropertiesToMap();
+    });
+  };
+
+  _setContainerRef = (element: HTMLDivElement) => {
+    this._containerRef = element;
+  };
+
+  async _syncImageToMap() {
+    if (this._isMounted && this.state.map) {
+      const map = this.state.map;
+      const { svg, cutoff, radius, isSvgInvalid } = this.props;
+      if (!svg || isSvgInvalid) {
+        map.setLayoutProperty('icon-layer', 'visibility', 'none');
+        return;
+      }
+      const imageData = await createSdfIcon({ svg, cutoff, radius });
+      if (map.hasImage(IconPreview.iconId)) {
+        // @ts-expect-error
+        map.updateImage(IconPreview.iconId, imageData);
+      } else {
+        map.addImage(IconPreview.iconId, imageData, {
+          sdf: true,
+          pixelRatio: CUSTOM_ICON_PIXEL_RATIO,
+        });
+      }
+      map.setLayoutProperty('icon-layer', 'icon-image', IconPreview.iconId);
+      map.setLayoutProperty('icon-layer', 'icon-size', 6);
+      map.setLayoutProperty('icon-layer', 'visibility', 'visible');
+      this._syncPaintPropertiesToMap();
+    }
+  }
+
+  _syncPaintPropertiesToMap() {
+    const { map, iconColor } = this.state;
+    if (!map) return;
+    map.setPaintProperty('icon-layer', 'icon-halo-color', '#000000');
+    map.setPaintProperty('icon-layer', 'icon-halo-width', 1);
+    map.setPaintProperty('icon-layer', 'icon-color', iconColor);
+    map.setLayoutProperty('icon-layer', 'icon-size', 12);
+  }
+
+  _initResizerChecker() {
+    this._checker = new ResizeChecker(this._containerRef!);
+    this._checker.on('resize', () => {
+      if (this.state.map) {
+        this.state.map.resize();
+      }
+    });
+  }
+
+  async _createMapInstance(): Promise<MapboxMap> {
```

Comment:
_createMapInstance does not need to be async. You are resolving the promise outside of the 'load' callback.